File: DynamicActions/Models/DynamicAction.cs
namespace DynamicActions.Models
{
    public class DynamicAction
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Expression { get; set; }
        public DynamicActionType DynamicActionType { get; set; }

        public DateTime Created { get; set; }
    }

    public enum DynamicActionType
    {
        Numeric,
        Text
    }
}

File: dynamic-actions-app/src/shared/models/dynamic-action.ts


export interface DynamicAction {
    id? : string; 
    expression : string; 
    name: string;
    created?: Date;
    dynamicActionType: string;
}

export interface DynamicActionCalculate{
    x: string;
    y: string;
    dynamicActionId: string;
}

export interface DynamicActionHistory {
    x: string;
    y: string;
    result: string;
    created: Date;
}

export interface DynamicActionCalculateResult{
    result: string;
    max: number;
    min: number;
    avg: number;
    lastMonth: number;
}


